Patch obtained from gitlab, should be able to remove upon next
release.

https://gitlab.isc.org/isc-projects/bind9/-/issues/5622

*** a/lib/dns/validator.c	Wed Nov  5 11:45:36 2025
--- b/lib/dns/validator.c	Wed Nov  5 11:45:36 2025
***************
*** 1093,1099 ****
  				goto done;
  			}
  			dst_key_free(&val->key);
! 		} else {
  			break;
  		}
  		dns_rdata_reset(&rdata);
--- 1093,1105 ----
  				goto done;
  			}
  			dst_key_free(&val->key);
! 		} else if (result != DST_R_UNSUPPORTEDALG) {
! 			/*
! 			 * We can encounter unsupported algorithm when the zone
! 			 * is signed with both supported and unsupported
! 			 * algorithm at the same time.  Stop looking in all
! 			 * other failure cases.
! 			 */
  			break;
  		}
  		dns_rdata_reset(&rdata);

